File: src/my_class_pkg/src/ros_server.cpp
#include "ros/ros.h"
#include "my_class_pkg/MyServiceMsg.h"

bool myServiceCallback(my_class_pkg::MyServiceMsg::Request &req,
                       my_class_pkg::MyServiceMsg::Response &res)
{
    res.output = req.input * 2;
    ROS_INFO("Request: input = %ld, output = %ld", (long)req.input, (long)res.output);
    return true;
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "my_service_node");
    ros::NodeHandle nh;
    ros::ServiceServer service = nh.advertiseService("my_service", myServiceCallback);
    ROS_INFO("Ready to receive service requests.");
    ros::spin();
    return 0;
}

File: src/my_class_pkg/scripts/ros_server.py
#!/usr/bin/env python3
import rospy
from my_class_pkg.srv import MyServiceMsg, MyServiceMsgResponse

def handle(req):
    rospy.loginfo("Service request: %d", req.input)
    return MyServiceMsgResponse(output=req.input * 2)

if __name__ == "__main__":
    rospy.init_node("my_service_server")
    srv = rospy.Service("my_service", MyServiceMsg, handle)
    rospy.loginfo("Ready to handle requests")
    rospy.spin()
